date_range/models/date_range_search_mixin.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2021 Opener B.V. <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from lxml import etree
+from psycopg2 import sql
+
+from odoo.osv.expression import (
+    NEGATIVE_TERM_OPERATORS, TRUE_DOMAIN, FALSE_DOMAIN
+)
+from odoo import _, api, fields, models
+
+
+class DateRangeSearchMixin(models.AbstractModel):
+    _name = "date.range.search.mixin"
+    _description = "Mixin class to add a Many2one style period search field"
+    # Override this attribute to apply the mixin to another field
+    _date_range_search_field = "date"
+    # Set this attribute to assign a date range of the type with this code.
+    # This has to be a date range type without company set
+    _date_range_assign_type_code = None
```

Comment:
for making this perfectly general, can't we assume `company_id` on the record and allow to fine tune the field name the way we do it here?

Reply:
What do you mean? I think having a company on the date range type is a problem because the attribute by necessity lacks a company context.

Reply:
but you can join the company field on the table to the range's company_id in the statement below, right?

Reply:
I'm just thinking that setting a company_id on date ranges by default was not the best design to begin with, so I am hesitant to go along with it. Or is that my lack of imagination?

Reply:
I'd think of a join condition like `{company_id_field} = res_company.id or {company_id_field} is null`

Reply:
Yes, I will add that if the feature is accepted.